tools/src/main/java/com/nedap/archie/rules/evaluation/evaluators/FunctionUtil.java

```diff
@@ -22,17 +22,20 @@ public static double castToDouble(Value value) throws FunctionCallException {
         throw new FunctionCallException("cannot cast " + value.getValue().getClass() + " to a number");
     }
 
-    public static int checkEqualLength(List<ValueList> arguments) {
-        int length = 1;
-        boolean first = true;
-        for(ValueList list:arguments) {
-            if(first && list.size() != 1) {
-                length = list.size();
-            } else  if(list.size() != 1 && list.size() != length) {
-                return -1;
+    /**
+     * All arguments should have an equal list size OR list size of allowedLength
+     */
+    public static int checkEqualLengthOrOne(List<ValueList> arguments) {
+        int[] allowedLengths = { 1, -1 };
```

Comment:
I find the array with two indexes much more confusing than a constant and a variable.